### PR TITLE
ruby 3.0.0

### PR DESCRIPTION
**Pull request by SeekingMeaning**
Created with `brew bump-formula-pr`.

~~`resource` blocks may require updates.~~

### CODE REVIEW

**Comment by carlocab** [comment]
For cucumber-cpp: https://github.com/Homebrew/homebrew-core/pull/67689

**Comment by SeekingMeaning** [comment]
11:

```
Error: 13 failed steps!
brew fetch --retry cucumber-cpp
brew linkage --test fastlane
brew test --retry --verbose fastlane
brew linkage --test libstfl
brew linkage --test macvim
brew test --retry --verbose macvim
brew linkage --test mikutter
brew test --retry --verbose mikutter
brew test --retry --verbose terraform_landscape
brew linkage --test travis
brew linkage --test vim
brew test --retry --verbose vim
brew linkage --test weechat
```

10.15:

```
Error: 12 failed steps!
brew linkage --test fastlane
brew test --retry --verbose fastlane
brew linkage --test libstfl
brew linkage --test macvim
brew test --retry --verbose macvim
brew linkage --test mikutter
brew test --retry --verbose mikutter
brew test --retry --verbose terraform_landscape
brew linkage --test travis
brew linkage --test vim
brew test --retry --verbose vim
brew linkage --test weechat
```

10.14:

```
Error: 17 failed steps!
brew linkage --test cocoapods
brew fetch --retry cucumber-cpp
brew linkage --test cucumber-ruby
brew test --retry --verbose cucumber-ruby
brew linkage --test fastlane
brew test --retry --verbose fastlane
brew linkage --test libstfl
brew linkage --test macvim
brew test --retry --verbose macvim
brew linkage --test mikutter
brew test --retry --verbose mikutter
brew linkage --test passenger
brew test --retry --verbose terraform_landscape
brew linkage --test travis
brew linkage --test vim
brew test --retry --verbose vim
brew linkage --test weechat
```

**Comment by SeekingMeaning** [comment]
`brew test terraform_landscape`:

```
/usr/local/Cellar/terraform_landscape/0.3.3/libexec/gems/treetop-1.6.10/lib/treetop/compiler/grammar_compiler.rb:50:in `class_eval': (eval):293: _1 is reserved for numbered parameter (SyntaxError)
(eval):305: _2 is reserved for numbered parameter
(eval):313: _3 is reserved for numbered parameter
(eval):340: _1 is reserved for numbered parameter
(eval):352: _2 is reserved for numbered parameter
```

**Comment by carlocab** [comment]
`cucumber-cpp` only depends on `ruby` for the test, so a revision bump is actually unnecessary there. It should be fixed here now thouhg.

**Comment by SeekingMeaning** [comment]
I actually bumped `cucumber-ruby` (not `cucumber-cpp`)

**Comment by carlocab** [comment]
Right, my mistake. I forgot we had two `cucumber-` formulae. 😄

**Comment by SeekingMeaning** [comment]
11.0, 10.14:

```
Error: 6 failed steps!
brew test --retry --verbose terraform_landscape
brew install --verbose --build-bottle fastlane
brew install --verbose --build-bottle macvim
brew install --verbose --build-bottle mikutter
brew install --verbose --build-bottle vim
brew install --verbose --build-bottle weechat
```

10.15:

```
Error: 7 failed steps!
brew test --retry --verbose terraform_landscape
brew install --verbose --build-bottle cucumber-ruby
brew install --verbose --build-bottle fastlane
brew install --verbose --build-bottle macvim
brew install --verbose --build-bottle mikutter
brew install --verbose --build-bottle vim
brew install --verbose --build-bottle weechat
```

**Comment by carlocab** [comment]
https://github.com/Homebrew/homebrew-core/blob/d1bdde1ebb3fd3bb28475ef4cb744995dfaf353d/Formula/cucumber-ruby.rb#L18

`s/catalina/big_sur`? It's possible cucumber-ruby just doesn't like Catalina Ruby. (See also CI forcing vendor Ruby on Catalina.)

**Comment by SeekingMeaning** [comment]
> (See also CI forcing vendor Ruby on Catalina.)

Oh right, I forgot we're doing that

> `s/catalina/big_sur`?

Thanks, I'll try that

**Review by chenrui333 — DISMISSED** [review]
🎉

File: Formula/fastlane.rb [hunk 1]
```diff
@@ -19,7 +20,7 @@ class Fastlane < Formula
     sha256 "b9e0be03c7c28d5d11d274ae8c0b20a3d96e5619d9bdb6fca788d312a2365070" => :mojave
   end
 
-  depends_on "ruby"
+  depends_on "ruby@2.6"
```
Comment (chenrui333) [inline Formula/fastlane.rb:23]: retrospectively thinking, I wonder if we should add ruby@2.7 and have the formula depends on it.

**Comment by kemchenj** [comment]
I got an error from `brew upgrade ruby`, here is the output:

```
Updating Homebrew...
Updating Homebrew...
==> Upgrading 1 outdated package:
ruby 3.0.0
==> Upgrading ruby -> 3.0.0
==> Downloading https://homebrew.bintray.com/bottles/ruby-3.0.0.big_sur.bottle.tar.gz
Already downloaded: /Users/kemchenj/Library/Caches/Homebrew/downloads/f12bffe965b0c6b03795b55c04f3406881f1e84ef085904e2b6bcc4d66bcd7b8--ruby-3.0.0.big_sur.bottle.tar.gz
==> Pouring ruby-3.0.0.big_sur.bottle.tar.gz
Warning: The post-install step did not complete successfully
You can try again using `brew postinstall ruby`
Error: Failure while executing; `/usr/local/opt/ruby/bin/ruby -e print\ Gem.ruby_api_version` was terminated by uncaught signal ILL. Here's the output:
```

`brew config`:

```
HOMEBREW_VERSION: 2.7.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: 993bf95877143afe1cede022083499e0b4fe336c
Last commit: 7 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: be000dc14198482f6f457a13cefde2ee11e56930
Core tap last commit: 7 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_BROWSER: open
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 8
all_proxy: socks5://127.0.0.1:6153
http_proxy: http://127.0.0.1:6152
https_proxy: http://127.0.0.1:6152
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit westmere
Clang: 12.0 build 1200
Git: 2.24.3 => /Applications/Xcode-12.3.app/Contents/Developer/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.1-x86_64
CLT: N/A
Xcode: 12.3 => /Applications/Xcode-12.3.app/Contents/Developer
Rosetta 2: true
```

**Comment by carlocab** [comment]
I think this is the same issue: https://github.com/Homebrew/homebrew-core/issues/67713

The `ruby` formula will probably have to be rebuilt. In the meantime, you should be able to do `brew install -s ruby`.

**Comment by kemchenj** [comment]
`brew install -s ruby` works, thanks for your reply.

**Comment by fxcoudert** [comment]
@SeekingMeaning I'd like to understand, if ruby needs the `-fdeclspec` flag on clang (and clang only: https://github.com/Homebrew/homebrew-core/pull/67910, https://github.com/Homebrew/homebrew-core/pull/67901 and https://github.com/Homebrew/homebrew-core/pull/67898) why doesn't the build system set it up? It sounds like a bug, has it been reported?

**Comment by SeekingMeaning** [comment]
> I'd like to understand, if ruby needs the `-fdeclspec` flag on clang (and clang only: #67910, #67901 and #67898) why doesn't the build system set it up?

What do you mean?

> It sounds like a bug, has it been reported?

No, I have not

**Comment by fxcoudert** [comment]
> What do you mean?

I mean when you build vim, you don't really expect to have to tell it what compiler flags the ruby bindings need… it should know. Whether it includes that information in its build system, or it calls some ruby-provided mechanism to determine it (pkg-config, a `foo-config` script, …) I don't know, because I don't know Ruby, really

**Comment by fxcoudert** [comment]
Ah, they've added it: https://github.com/vim/vim/commit/864a28b6a6ae4f1a56e230be26bc6d13e3f8b3d9
Could you report the issue to macvim and weechat, and make a note in the formulas to remove those `CFLAGS` in the next version?

**Pull request closed without merging**